### ISSUE TITLE
Replace use of sizeof() in file import/export code with explicit byte count

### ISSUE THREAD

**Issue by imaami** [pjreddie/darknet#78]
Currently the binary layout of the weights files (and possibly other files) is defined using `sizeof(int)` and `sizeof(size_t)` which aren't guaranteed to be specific numbers across platforms. While `sizeof(int)` is most likely 4 bytes on any machine that Darknet is built for today, `sizeof(size_t)` can be either 4 or 8 bytes simply depending on whether the compiler is invoked with `-m32` or `-m64`.

A binary file format should definitely be predictable, but unfortunately anyone who downloads the pre-trained weights files and attempts to run Darknet on a machine with a different build config than the author's will likely experience unexplainable behavior.

**Comment by imaami** [comment]
@adroit91 You're right. I apologize that I haven't written a patch yet. I'll try to do that soon (and I hope @pjreddie starts merging my pull requests, there are several waiting already...)

**Comment by zinkcious** [comment]
Did @pjreddie  accept your pull requests? @imaami
